### DISCUSSION TITLE
toggle  global light and dark mode?

### DISCUSSION THREAD

**Discussion by erflot** [emilk/egui#148] [Q&A] [answered]
Hi,
i made a copy named light() from default() [impl Default for Widgets in style.rs] and changed colors inside light() to mimic a global light mode.
Where to place light() for toggling globally between light() and default() for all widgets? 
Radio buttons in a SidePanel should trigger the switch.
Thanks for any hints.

**Comment by emilk** [comment]
Oh, a light mode! Nice, I've been meaning to make my own one day. Please make a PR, or share it some other way!

To switch the global style, use [`ctx.set_style(style)`](https://docs.rs/egui/0.8.0/egui/struct.Context.html#method.set_style).

**Comment by erflot** [comment, marked as answer]
Sorry for not providing a full fledged light mode. I just made a very simple one to escape the low winter sun on the display.
Thank you very much for the hint, it works as supposed:
```rust           
            if ui.button("dark<>light").clicked {
                let mut style = ui.style().clone();
                if style.visuals.widgets.noninteractive.bg_fill == Color32::from_gray(30) {
                    style.visuals.widgets.noninteractive.bg_fill = Color32::WHITE;
                    style.visuals.widgets.noninteractive.fg_stroke = Stroke { width: 1.0, color: Color32::BLACK };
                    style.visuals.widgets.active.bg_fill = Color32::WHITE;
                    style.visuals.widgets.active.fg_stroke = Stroke { width: 1.0, color: Color32::BLACK };
                    style.visuals.widgets.inactive.bg_fill = Color32::WHITE;
                    style.visuals.widgets.inactive.fg_stroke = Stroke { width: 1.0, color: Color32::BLACK };
                    style.visuals.widgets.hovered.bg_fill = Color32::WHITE;
                    style.visuals.widgets.hovered.fg_stroke = Stroke { width: 1.0, color: Color32::BLACK };
                } else {
                    style = Style::default();
                }
                ctx.set_style(style);
            }

**Reply by emilk** [reply]
I decided to spend a few hours working on a proper light mode theme. Coming in egui 0.9:

<img width="361" alt="light_mode_preview" src="https://user-images.githubusercontent.com/1148717/106803903-f948e980-6664-11eb-986f-33e4543bb37b.png">

**Reply by erflot** [reply]
Yes, thanks, it looks completely professional. Your work was worth it. Persistence of the dark/light setting would make another egui highlight.

**Reply by mjehrhart** [reply]
Is there a way to set a light or dark theme on master branch?

**Reply by BctfN0HUK7Yg** [reply]
ui.ctx().set_visuals(Visuals::dark());

**Reply by mjehrhart** [reply]
ctx.set_visuals(egui::Visuals::light());

**Reply by mjehrhart** [reply]
thanks BctfN0HUK7Yg, I had just found it when I notice your comment. Thanks.

**Comment by ShayBox** [comment]
Here's a more updated example using the built-in dark/light themes
```rust
if ui.button("Toggle Dark/Light Mode").clicked() {
    let visuals = if ui.visuals().dark_mode {
        Visuals::light()
    } else {
        Visuals::dark()
    };
    ctx.set_visuals(visuals);
}
```

**Comment by WojciechMula** [comment]
The standardized way is to use either `egui::widgets::global_dark_light_mode_buttons` or `egui::widgets::global_dark_light_mode_switch`. Docs: https://docs.rs/egui/latest/egui/?search=global
